src/test/java/teammates/test/cases/ui/pagedata/StudentCommentsPageDataTest.java

```diff
@@ -0,0 +1,339 @@
+        String actualCourseId = data.getCourseId();
+        assertEquals(expectedCourseId, actualCourseId);
+        
+        String expectedCourseName = courseName;
+        String actualCourseName = data.getCourseName();
+        assertEquals(expectedCourseName, actualCourseName);
+        
+        int index = coursePaginationList.indexOf(courseId);
+        String expectedPreviousPageLink = index > 0 ? coursePaginationList.get(index - 1) : "javascript:;";
+        String expectedNextPageLink = 
+                index < coursePaginationList.size() - 1 ? coursePaginationList.get(index + 1) : "javascript:;";
+        List<String> expectedCoursePaginationList = coursePaginationList;
+        String expectedActiveCourse = courseId;
+        String expectedLink = Const.ActionURIs.STUDENT_COMMENTS_PAGE;
+        expectedLink = Url.addParamToUrl(expectedLink, Const.ParamsNames.USER_ID, account.googleId);
+        CoursePagination actualCoursePagination = data.getCoursePagination();
+        assertEquals(expectedPreviousPageLink, actualCoursePagination.getPreviousPageLink());
+        assertEquals(expectedNextPageLink, actualCoursePagination.getNextPageLink());
+        assertEquals(expectedCoursePaginationList, actualCoursePagination.getCoursePaginationList());
+        assertEquals(expectedActiveCourse, actualCoursePagination.getActiveCourse());
+        assertEquals(expectedLink, actualCoursePagination.getUserCommentsLink());
+    }
+
+    private static void checkCommentsForStudentsTablesEqual(CommentsForStudentsTable expectedCommentsForStudentsTable,
+                                                            CommentsForStudentsTable actualCommentsForStudentsTable) {
+        assertEquals(expectedCommentsForStudentsTable.getGiverDetails(),
+                     actualCommentsForStudentsTable.getGiverDetails());
+        assertEquals(expectedCommentsForStudentsTable.getExtraClass(), actualCommentsForStudentsTable.getExtraClass());
+        List<Comment> actualCommentRows = actualCommentsForStudentsTable.getRows();
+        List<Comment> expectedCommentRows = expectedCommentsForStudentsTable.getRows();
+        assertEquals(expectedCommentRows.size(), actualCommentRows.size());
+        
+        for(int i = 0; i < expectedCommentRows.size(); i++) {
+            checkCommentRowsEqual(expectedCommentRows.get(i), actualCommentRows.get(i));
+        }
+    }
+
+    private static void checkFeedbackSessionRowsEqual(
+            FeedbackSessionRow expectedFeedbackSessionRow, FeedbackSessionRow actualFeedbackSessionRow) {
+        assertEquals(expectedFeedbackSessionRow.getFeedbackSessionName(), 
+                     actualFeedbackSessionRow.getFeedbackSessionName());
+        assertEquals(expectedFeedbackSessionRow.getCourseId(), actualFeedbackSessionRow.getCourseId());
+        assertEquals(expectedFeedbackSessionRow.getQuestionTables().size(), 
+                     actualFeedbackSessionRow.getQuestionTables().size());
```

Comment:
Do this comparison after you've extracted `expected/actualQuestionTables` to their own variables, like you did up there in `expected/actualCommentRows`.